packages/svelte/src/animate/index.js

```diff
@@ -16,13 +16,21 @@ export function flip(node, { from, to }, params = {}) {
 
 	var transform = style.transform === 'none' ? '' : style.transform;
 	var [ox, oy] = style.transformOrigin.split(' ').map(parseFloat);
+
+	var m = new DOMMatrixReadOnly(transform);
+
 	var dsx = from.width / to.width;
 	var dsy = from.height / to.height;
 
 	var dx = (from.left + dsx * ox - (to.left + ox)) / zoom;
 	var dy = (from.top + dsy * oy - (to.top + oy)) / zoom;
 	var { delay = 0, duration = (d) => Math.sqrt(d) * 120, easing = cubicOut } = params;
 
+	var rf = Math.atan2(m.m21, m.m11) + Math.atan2(from.bottom - from.top, from.right - from.left);
```

Comment:
What about validating this calculation like this:

```js
 function get_initial_angle(transform) {
	if (typeof transform === 'string') return 0

	var m = new DOMMatrixReadOnly(transform);

	return Math.atan2(m.m21, m.m11);
 }
 
```
```js
	 var rf = get_initial_angle(transform) + Math.atan2(from.bottom - from.top, from.right - from.left);
```

I think skipping these calculations if transform is none would be cool when multiple elements without initial transforms are animated but I'm unsure how much it would impact complex transformations